src/component/application/ApplicationEdit.tsx

```diff
@@ -0,0 +1,154 @@
+/* eslint react/no-multi-comp:off */
+import { useContext, useEffect, useState } from 'react';
+import {
+    Avatar,
+    Link,
+    Icon,
+    IconButton,
+    Button,
+    LinearProgress,
+    Typography,
+} from '@material-ui/core';
+import { Link as LinkIcon } from '@material-ui/icons';
+import ConditionallyRender from '../common/ConditionallyRender/ConditionallyRender';
+import {
+    formatDateWithLocale,
+} from '../common/util';
+import { UPDATE_APPLICATION } from '../providers/AccessProvider/permissions';
+import ApplicationView from './ApplicationView';
+import ApplicationUpdate from './ApplicationUpdate';
+import TabNav from '../common/TabNav/TabNav';
+import Dialogue from '../common/Dialogue';
+import PageContent from '../common/PageContent';
+import HeaderTitle from '../common/HeaderTitle';
+import AccessContext from '../../contexts/AccessContext';
+import useApplicationsApi from '../../hooks/api/actions/useApplicationsApi/useApplicationsApi';
+import useApplication from '../../hooks/api/getters/useApplication/useApplication';
+import { useHistory, useParams } from 'react-router-dom';
+import { useLocationSettings } from '../../hooks/useLocationSettings';
+
+const ApplicationEdit = () => {
+    const history = useHistory();
+    const { name } = useParams<{ name: string }>();
+    const { refetchApplication, application } = useApplication(name);
+    const { appName, url, description, icon = 'apps', createdAt } = application;
+    const { hasAccess } = useContext(AccessContext);
+    const { deleteApplication } = useApplicationsApi();
+    const { locationSettings } = useLocationSettings();
+
+    const [loading, setLoading] = useState(true);
```

Comment:
This should be uneccesary, you can get the loading state from useApplication hook directly.